proto/rpc.proto

```diff
@@ -0,0 +1,43 @@
+syntax = "proto3";
+
+import "block.proto";
+import "digest.proto";
+
+// Request a single block header by its block number.
+message GetBlockHeaderByNumberRequest {
+    uint64 block_number = 1;
+}
+
+// Request a single block header by its hash.
+message GetBlockHeaderByHashRequest {
+    Digest block_hash = 1;
+}
+
+// Request a sequence of block headers by block number using the range [start,end].
+message GetBlockHeaderByNumberRangeRequest {
+    // The first block header to include in the response. If the value is not
+    // provided start at the genesis block.
+    optional uint64 start = 1;
+
+    // The last block header to include in the response. If the value is not
+    // provided the server will use the last known block.
+    optional uint64 end = 2;
+}
+
+message GetBlockHeaderByNumberResponse {
+    optional BlockHeader block_header = 1;
+}
+
+message GetBlockHeaderByHashResponse {
+    optional BlockHeader block_header = 1;
+}
+
+message GetBlockHeaderByNumberRangeResponse {
+    optional BlockHeader block_header = 1;
+}
+
+service Api {
+    rpc GetBlockHeaderByNumber(GetBlockHeaderByNumberRequest) returns (GetBlockHeaderByNumberResponse) {}
+    rpc GetBlockHeaderByHash(GetBlockHeaderByHashRequest) returns (GetBlockHeaderByHashResponse) {}
```

Comment:
I'm not yet sure we need this as a part of the basic API. Should we exclude it for now?

Reply:
removed